include/d/d_rumble.h

```diff
@@ -0,0 +1,100 @@
+#ifndef D_RUMBLE_H
+#define D_RUMBLE_H
+
+#include "common.h"
+
+class dRumbleEntry_c {
+    friend class dRumble_c;
+
+public:
+    dRumbleEntry_c(s32 length, u32 bits, f32 instesity);
+
+private:
+    /* 0x00 */ s32 mLength;
+    /* 0x04 */ s32 mBits;
+    /* 0x08 */ f32 mIntensity;
+};
+
+class dRumbleIdx_c {
+public:
+    ~dRumbleIdx_c() {
+        stop();
+    }
+
+    /* Queues the entry into the Rumble Manager */
+    bool start(const dRumbleEntry_c &entry, u32 flags);
+    void stop();
+
+    bool isActive() const {
+        return mIdx != -1;
+    }
+    void setIdx(s32 idx) {
+        mIdx = idx;
+    }
+    s32 getIdx() const {
+        return mIdx;
+    }
+
+private:
+    s32 mIdx;
+};
+
+/**
+ * Made-up name.
+ * Size: 0x54
+ */
+class dRumble_c {
+public:
+    enum Flags_c {
+        FLAG_SLOT0 = 0x1,
+        FLAG_SLOT1 = 0x2,
+        FLAG_ACTIVE = 0x10,
+        FLAG_INITIALIZE = 0x200,
+    };
+
+    struct RumbleData {
+        /* 0x00 */ s32 mLength;
+        /* 0x04 */ s32 mBitsLeft;
+        /* 0x08 */ s32 mRumbleBits;
+        /* 0x0C */ u32 mFlags;
+        /* 0x10 */ f32 mIntensity;
+    };
+
+public:
+    /** Generates the instance */
+    static void init();
+    /** Deletes the instance */
+    static void deinit();
+
+    /** Runs the rumble instances */
+    static void play();
```

Comment:
Alternatively `create`, `remove`, `execute` for consistency with other systems in the main DOL?